azurerm/resource_arm_mssql_server_security_alert_policy.go

```diff
@@ -0,0 +1,289 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/2017-03-01-preview/sql"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMssqlServerSecurityAlertPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMssqlServerSecurityAlertPolicyCreateUpdate,
+		Read:   resourceArmMssqlServerSecurityAlertPolicyRead,
+		Update: resourceArmMssqlServerSecurityAlertPolicyCreateUpdate,
+		Delete: resourceArmMssqlServerSecurityAlertPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"server_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateMsSqlServerName,
+			},
+
+			"disabled_alerts": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      schema.HashString,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"Sql_Injection",
+						"Sql_Injection_Vulnerability",
+						"Access_Anomaly",
+						"Data_Exfiltration",
+						"Unsafe_Action",
+					}, true),
+				},
+			},
+
+			"email_account_admins": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"email_addresses": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Set: schema.HashString,
+			},
+
+			"retention_days": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      0,
+				ValidateFunc: validation.IntAtLeast(0),
+			},
+
+			"state": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
How come we are ignoring case here? is the API returning the wrong casing compared to the constants in the SDK?

Reply:
When I remember correctly this was the issue back then. Now it doesn't seem to be a problem anymore, removed it.